pkg/querybuilderfilters/model_plan_section_filter.go

```diff
@@ -0,0 +1,76 @@
+package querybuilderfilters
+
+import (
+	"fmt"
+
+	"github.com/aquasecurity/esquery"
+
+	"github.com/cmsgov/mint-app/pkg/graph/model"
+	"github.com/cmsgov/mint-app/pkg/shared/oddsearch"
+)
+
+// ModelPlanSectionFilter is a filter for the model plan section filter
+type ModelPlanSectionFilter struct {
+	qb                *oddsearch.QueryBuilder
+	sectionTableIDMap map[model.SearchableTaskListSection]int
+}
+
+// NewModelPlanSectionFilter creates a new ModelPlanSectionFilter
+func NewModelPlanSectionFilter(qb *oddsearch.QueryBuilder) *ModelPlanSectionFilter {
+	return &ModelPlanSectionFilter{
+		qb: qb,
+		sectionTableIDMap: map[model.SearchableTaskListSection]int{
+			model.SearchableTaskListSectionBasics:                          2,
+			model.SearchableTaskListSectionGeneralCharacteristics:          8,
+			model.SearchableTaskListSectionParticipantsAndProviders:        11,
+			model.SearchableTaskListSectionBeneficiaries:                   3,
+			model.SearchableTaskListSectionOperationsEvaluationAndLearning: 10,
+			model.SearchableTaskListSectionPayment:                         12,
+		},
+	}
+}
+
+// HandleFilter handles the filter value for the ModelPlanSectionFilter
+func (f *ModelPlanSectionFilter) HandleFilter(value interface{}) error {
+	sectionQuery := esquery.Bool()
+
+	switch v := value.(type) {
+	case []interface{}:
+		for _, val := range v {
```

Comment:
Do you think that table_name should have this treatment as well? It probably can be added on later if needed. Right now, I couldn't specifically request plan_documents and model_plan changes only for example (because they aren't part of the task list, and putting in two table name parameter returns an empty array.)